Plugins/ConnectMocksPlugin/ConnectMockGenerator.swift

```diff
@@ -219,4 +224,28 @@ private extension MethodDescriptor {
             """
         }
     }
+
+    func callbackAvailabilityAnnotation() -> String? {
+        if self.options.deprecated {
+            // swiftlint:disable line_length
+            return """
+            @available(iOS, introduced: 12, deprecated: 12, message: "This function has been marked deprecated.")
+            @available(macOS, introduced: 10.15, deprecated: 10.15, message: "This function has been marked deprecated.")
+            @available(tvOS, introduced: 13, deprecated: 13, message: "This function has been marked deprecated.")
+            @available(watchOS, introduced: 6, deprecated: 6, message: "This function has been marked deprecated.")
+            """
+            // swiftlint:enable line_length
+        } else {
+            return nil
+        }
+    }
+
+    func asyncAwaitAvailabilityAnnotation() -> String? {
+        if self.options.deprecated {
+            // swiftlint:disable:next line_length
+            return "@available(iOS, introduced: 13, deprecated: 13, message: \"This function has been marked deprecated.\")"
+        } else {
+            return nil
+        }
+    }
```

Comment:
This function signature and behavior differs from its sibling defined in `ConnectClientGenerator` since the entire type is already annotated with an `@available(iOS 13, *)` in the generated mock. Because of that existing annotation, we only need to produce something if in the case that it's more restrictive than the availability rules defined at the higher scope.